### ISSUE TITLE
[Bug]: Cannot close menu/dropdown by clicking iframe

### ISSUE THREAD

**Issue by horans** [tailwindlabs/headlessui#707]
### What package within Headless UI are you using?

@headlessui/vue

### What version of that package are you using?

v1.4.0

### What browser are you using?

Chrome v92

### Reproduction repository

https://headlessui.dev/vue/menu

### Describe your issue

When a menu/dropdown is open, it cannot be closed by interacting(click) with iframe on the page.

Reproduction:
1. Click left top react/vue menu to open
2. Click right preview area (which is a iframe)

Expecting: close/blur menu
Result: menu stay as open/focus

Reference: <https://gist.github.com/jaydson/1780598>

**Comment by biesbjerg** [comment]
Maybe this can help someone, until a proper fix is implemented:

```vue
<template>
  <Listbox>
    <ListboxButton ref="buttonRef">..</ListboxButton>
    <ListboxOptions ref="optionsRef">...</ListboxOptions>
  </Listbox>
</template>

<script lang="ts">
import { Listbox, ListboxButton, ListboxOption, ListboxOptions } from "@headlessui/vue";
import { defineComponent, nextTick, onMounted, onUnmounted, ref } from "vue";

export default defineComponent({
  setup() {
    const buttonRef = ref<typeof ListboxButton>();
    const optionsRef = ref<typeof ListboxOptions>();

    const closeOnIframeClick = () => {
      const activeElement = document.activeElement as HTMLElement;
      if (activeElement instanceof HTMLIFrameElement) {
        optionsRef.value?.el?.dispatchEvent(
          new KeyboardEvent("keydown", {
            key: "Escape"
          })
        );
        nextTick(() => {
          buttonRef.value?.el?.blur();
          activeElement.focus();
        });
      }
    };

    onMounted(() => {
      window.addEventListener("blur", closeOnIframeClick);
    });

    onUnmounted(() => {
      window.removeEventListener("blur", closeOnIframeClick);
    });

    return {
      buttonRef,
      optionsRef,
    };
  }
});
</script>

**Comment by thecrypticace** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This should be fixed by #1552, and will be available in the next release.

You can already try it using npm install @headlessui/react@insiders.
